src/defrag.c

```diff
@@ -842,7 +834,7 @@ static int defragLaterItem(robj *ob, unsigned long *cursor, monotime endtime, in
             long long endtimeWallClock = ustime() + (endtime - getMonotonicUs());
             return moduleLateDefrag(sds_key_passed_as_robj, ob, cursor, endtimeWallClock, dbid);
         } else {
-            *cursor = 0; /* object type may have changed since we schedule it for later */
+            *cursor = 0; /* object type/encoding may have changed since we schedule it for later */
         }
     } else {
```

Comment:
This, and all the changes in `defrag.c` look good to me.  I'll defer to Viktor's comments re: the other refactoring.